src/logic/collectibles/store/actions/fetchCollectibles.ts

```diff
@@ -5,7 +5,11 @@ import { getConfiguredSource } from 'src/logic/collectibles/sources'
 import { addNftAssets, addNftTokens } from 'src/logic/collectibles/store/actions/addCollectibles'
 import { safeParamAddressFromStateSelector } from 'src/routes/safe/store/selectors'
 
+let isFetchingData = false
+
 const fetchCollectibles = () => async (dispatch, getState) => {
+  if (isFetchingData) return
+  isFetchingData = true
```

Comment:
Can you add a try/catch in this block? if something fails, like `fetchAllUserCollectiblesByCategoryAsync`, `isFetchingData` will remain as `true`.

Reply:
Yep good idea :)

Reply:
done